glotaran/project/test/test_project.py

```diff
@@ -0,0 +1,168 @@
+
+    model_folder = Path(project_folder) / "models"
+    assert model_folder.exists()
+
+    model_file = model_folder / "test_model.yml"
+    assert model_file.exists()
+
+    assert project.has_models
+
+    model = project.load_model("test_model")
+    assert "megacomplex_parallel_decay" in model.megacomplex
+
+
+@pytest.mark.parametrize("name", ["test_parameter", None])
+@pytest.mark.parametrize("fmt", ["yml", "yaml", "csv"])
+def test_generate_parameters(project_folder, project_file, name, fmt):
+    project = Project.open(project_file)
+
+    assert project.has_models
+
+    project.generate_parameters("test_model", name=name, fmt=fmt)
+
+    parameter_folder = Path(project_folder) / "parameters"
+    assert parameter_folder.exists()
+
+    parameter_file_name = f"{'test_model_parameters' if name is None else name}.{fmt}"
+    parameter_file = parameter_folder / parameter_file_name
+    assert parameter_file.exists()
+
+    assert project.has_parameters
+
+    model = project.load_model("test_model")
+    parameters = project.load_parameters("test_model_parameters" if name is None else name)
+
+    for parameter in model.get_parameter_labels():
+        assert parameters.has(parameter)
+    parameter_file.unlink()
+
+
+@pytest.mark.parametrize("name", ["test_data", None])
+def test_import_data(project_folder, project_file, test_data, name):
+    project = Project.open(project_file)
+
+    project.import_data(test_data, name=name)
+
+    data_folder = Path(project_folder) / "data"
+    assert data_folder.exists()
+
+    data_file_name = f"{'dataset_1' if name is None else name}.nc"
+    data_file = data_folder / data_file_name
+    assert data_file.exists()
+
+    assert project.has_data
+
+    data = project.load_data("dataset_1" if name is None else name)
+    assert data == example_dataset
+
+
+def test_create_scheme(project_folder, project_file):
+    project = Project.open(project_file)
+
+    project.generate_parameters("test_model", name="test_parameters")
+    scheme = project.create_scheme(
+        model="test_model", parameters="test_parameters", maximum_number_function_evaluations=1
+    )
+
+    assert "dataset_1" in scheme.data
+    assert "dataset_1" in scheme.model.dataset
+    assert scheme.maximum_number_function_evaluations == 1
+
+
+@pytest.mark.parametrize("name", ["test_run", None])
+def test_run_optimization(project_folder, project_file, name):
+    project_folder = Path(project_folder)
+    project = Project.open(project_file)
+
+    model_file = project_folder / "models" / "sequential.yml"
+    model_file.write_text(example_model_yml)
+
+    parameters_file = project_folder / "parameters" / "sequential.csv"
+    save_parameters(example_parameter, parameters_file)
+
+    data_folder = project_folder / "data"
+    assert data_folder.exists()
+    data_file = data_folder / "dataset_1.nc"
+    data_file.unlink()
+    save_dataset(example_dataset, data_file)
```

Comment:
Why do we need to delete and then save files over and over again?
I guess rewriting the fixtures with `tmp_path` would save us some headaches also, especially with parametrized tests it makes debugging easier since the parameters get appended to the folder name (here something like `test_run_optimization-test_run` and `test_run_optimization-None`)

Reply:
Well, the simple thing is that I want to test optimization and I use the example sequential model. The deletion is because the test is parameterized. There can be exists checks added, new fixtures made or whatever, but it all adds complexity in it's own way and choose the KISS way.

NOTE: you cannot do this prior to tests, otherwise e.g. the initial `assert not project has_models` would fail.

If you find a more easy way you can implement it, I had no trouble debugging and I don't see where one would have troubles. Thus I leave it like this.

Reply:
Let's leave this for a later refactor maybe for now it is fine.